### PR TITLE
Add wash gem blog post

### PR DESCRIPTION
**Pull request by ekinanp**
Signed-off-by: Enis Inan <enis.inan@puppet.com>

### CODE REVIEW

File: docs/_posts/2019-11-14-wash-gem.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+end
+{% endhighlight %}
+
+Notice that
+
+* Each class extends the `Wash::Entry` class.
+* Entry schemas are declared via class-level helpers like `label`, `is_singleton`, and `parent_of`.
+* Each Wash method corresponds to an instance method in the class. For example, a report's `read` action is implemented by `Report#read`. Similarly, a PE instance's `list` action is implemented by `PEInstance#list`.
+* Entry attributes are declared via the `attributes` helper and can be set/accessed as instance-level fields. For example, a report's `mtime` and `meta` attributes are set in the constructor (`initialize` method).
+* The `state` helper lets you specify instance-level fields that should be serialized as part of the `state` key in the entry's JSON. The `wash gem` will set these fields after it reconstructs your entry. Thus, you can safely access those fields' values when implementing the entry's methods. For example, the `@pe_name` field in `Report` is initialized in the constructor and then referenced in `Report#read`.
```
Comment (MikaelSmith) [inline docs/_posts/2019-11-14-wash-gem.md]: This is the first place you talk about invoking the plugin as a script and serializing state. You also don't tie state to entry later. That feels like assuming that they read a significant part of the docs before this blog post.

I'd suggest moving the introduction of how plugin's are invoked (and introducing the concept of state) before walking through an excerpt of the implementation.

File: docs/_posts/2019-11-14-wash-gem.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+The `Wash::Entry` class makes all of this possible. It also comes with even more helpers that let you do other things like specify cache TTLs (see `Wash::Entry#cache_ttls`) or any prefetched methods (see `Wash::Entry#prefetch`).
+
+Recall that plugin script invocations adopt the following calling convention
+
+```
+<plugin_script> <method> <entry> <args...>
+```
+
+All plugin script invocations are handled by the `Wash.run` function. Here's how `Wash.run` is used in `puppetwash`
```
Comment (MikaelSmith) [inline docs/_posts/2019-11-14-wash-gem.md]: I'd call out more clearly that `puppetwash` here refers to the script, and that the call to `Wash.run` is added at the end of it to actually run things. I don't feel like that's quite clear from what you've said here.

**Review by MikaelSmith — CHANGES_REQUESTED** [review]

File: docs/_posts/2019-11-14-wash-gem.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
+---
+title: "Using the wash gem to write your external plugins"
+summary: "Shows how the wash gem can be used to write external plugins"
+author: Enis Inan
+---
+
+Wash ships with plugins that let you interact with your AWS, GCP, Docker, and Kubernetes resources. You can also write your own plugins via the [external plugin interface]({{ '/docs/external-plugins' | relative_url }}). If you're a Ruby developer, then the [wash gem](http://127.0.0.1:4000/wash/) makes this even easier by letting you model your entries as classes. We'll show you what this looks like by walking you through some of the [puppetwash](https://github.com/puppetlabs/puppetwash/tree/dadfbc3bb7b4f8fe379086182c3006237028ffc8) plugin's implementation, specifically the [puppetwash.rb](https://github.com/puppetlabs/puppetwash/blob/dadfbc3bb7b4f8fe379086182c3006237028ffc8/puppetwash.rb) file.
```
[the remark this answers is not in the document]
Comment (ekinanp) [reply docs/_posts/2019-11-14-wash-gem.md]: Yeah, let's tag puppetwash, that would be easier

File: docs/_posts/2019-11-14-wash-gem.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
+---
+title: "Using the wash gem to write your external plugins"
+summary: "Shows how the wash gem can be used to write external plugins"
+author: Enis Inan
```
Comment (MikaelSmith) [inline docs/_posts/2019-11-14-wash-gem.md]: Now that there's an authors section, you can reference it with `enisinan`.

**Comment by ekinanp** [comment]
Updated. I think I've addressed all the feedback.

File: docs/_posts/2019-11-14-wash-gem.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+Now let `<entry> = <path> <state>`. Then the plugin script’s usage becomes
+
+```
+<plugin_script> <method> <entry> <args...>
+```
+
+From this usage, we see that `<path>` and `<state>` are two different representations of an entry. `<path>` is useful for simple plugins where reconstructing the entry is easy. `<state>` is useful for more complicated plugins where entries could be represented as classes.
+
+If you're a Ruby developer, then the [wash gem](https://github.com/puppetlabs/wash-ruby/tree/0.1.2) makes external plugin development easy by letting you model your entries as classes. We'll show you what this looks like by walking you through some of the [puppetwash](https://github.com/puppetlabs/puppetwash/tree/dadfbc3bb7b4f8fe379086182c3006237028ffc8) plugin's implementation, specifically the [puppetwash.rb](https://github.com/puppetlabs/puppetwash/blob/dadfbc3bb7b4f8fe379086182c3006237028ffc8/puppetwash.rb) file.
```
Comment (MikaelSmith) [inline docs/_posts/2019-11-14-wash-gem.md]: This isn't a tagged version. Were you going to tag it?

**Review by MikaelSmith — APPROVED** [review]

File: docs/_posts/2019-11-14-wash-gem.md (hunk 5 above)
Comment (ekinanp) [reply docs/_posts/2019-11-14-wash-gem.md]: Is tagging all we need to release it?

Comment (MikaelSmith) [reply docs/_posts/2019-11-14-wash-gem.md]: Yeah, I think so.

**Pull request merged**